pkg/storer/migration/step_03.go

```diff
@@ -0,0 +1,158 @@
+		STEP 1, remove all of the BinItem entired
+		STEP 2, iterate ChunkBinItem, delete old ChunkBinItem and BatchRadiusItem
+		STEP 3, get new binID using IncBinID, create new ChunkBinItem and BatchRadiusItem
+	*/
+
+	rs := reserve.Reserve{}
+	logger := log.NewLogger("migration-step-03", log.WithSink(os.Stdout))
+	logger.Info("starting migration for reconstructing reserve bin IDs, do not interrupt or kill the process...")
+
+	// STEP 1
+	for i := uint8(0); i < swarm.MaxBins; i++ {
+		err := st.Delete(&reserve.BinItem{Bin: i})
+		if err != nil {
+			return err
+		}
+	}
+	logger.Info("removed all bin index entries")
+
+	var chunkBinItems []*reserve.ChunkBinItem
+	err := st.Iterate(
+		storage.Query{
+			Factory: func() storage.Item { return &reserve.ChunkBinItem{} },
+		},
+		func(res storage.Result) (bool, error) {
+			item := res.Entry.(*reserve.ChunkBinItem)
+			chunkBinItems = append(chunkBinItems, item)
+			return false, nil
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(chunkBinItems); i += 10000 {
+		end := i + 10000
+		if end > len(chunkBinItems) {
+			end = len(chunkBinItems)
+		}
+
+		b, err := st.Batch(context.Background())
+		if err != nil {
+			return err
+		}
+
+		for _, item := range chunkBinItems[i:end] {
+			err = b.Delete(item)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = b.Commit()
+		if err != nil {
+			return err
+		}
+	}
+	logger.Info("removed all bin ids", "total_entries", len(chunkBinItems))
+	chunkBinItems = nil
+
+	var batchRadiusItems []*reserve.BatchRadiusItem
+	err = st.Iterate(
+		storage.Query{
+			Factory: func() storage.Item { return &reserve.BatchRadiusItem{} },
+		},
+		func(res storage.Result) (bool, error) {
+			item := res.Entry.(*reserve.BatchRadiusItem)
+			batchRadiusItems = append(batchRadiusItems, item)
+
+			return false, nil
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	logger.Info("found reserve chunk entries, adding new entries", "total_entries", len(batchRadiusItems))
+
+	batchSize := 10000
+	missingChunks := 0
+
+	for i := 0; i < len(batchRadiusItems); i += batchSize {
+		end := i + batchSize
+		if end > len(batchRadiusItems) {
+			end = len(batchRadiusItems)
+		}
+
+		b, err := st.Batch(context.Background())
+		if err != nil {
+			return err
+		}
+
+		for _, item := range batchRadiusItems[i:end] {
+			hasChunkEntry, err := st.Has(&chunkstore.RetrievalIndexItem{Address: item.Address})
+			if err != nil {
+				return err
+			}
+
+			if !hasChunkEntry {
+				err = b.Delete(item)
+				if err != nil {
+					return err
+				}
+				missingChunks++
+			} else {
+				// STEP 3
+				newBinID, err := rs.IncBinID(st, item.Bin)
+				if err != nil {
+					return err
+				}
+
+				item.BinID = newBinID
+				err = b.Put(item)
+				if err != nil {
+					return err
+				}
+
+				err = b.Put(&reserve.ChunkBinItem{
+					BatchID:   item.BatchID,
+					Bin:       item.Bin,
+					Address:   item.Address,
+					BinID:     newBinID,
+					ChunkType: swarm.ChunkTypeContentAddressed,
```

Comment:
this has to come from the already existing item

Reply:
> this has to come from the already existing item

Okay, this is a problem. We will have to write some special function to fill this field in this case. The problem is the chunkBinItem may or may not be present. This is because the problematic entries would have overwritten the chunkBinItems. So even if we try to fill from the existing chunkBinItem and if its not found we will have issues.

Instead, this field is not used right now. So what we can do is update with the default value which is unspecified. And then have a separate function called in the storer.New function which looks at the chunkstore and populates this field properly for the ones that are unspecified.

Reply:
the problem with leaving it as default is the sampler has this check
```
				// Skip chunks if they are not SOC or CAC
				if chItem.Type != swarm.ChunkTypeSingleOwner &&
					chItem.Type != swarm.ChunkTypeContentAddressed {
					wstat.RogueChunk++
					continue
				}
```

Reply:
we can remove the field altogether and have the type computed on the fly wherever needed

Reply:
> 

So I figured out how to do it... check the latest diff